src/Symfony/Component/Scheduler/RecurringMessage.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Scheduler;
+
+use Symfony\Component\Scheduler\Trigger\CronExpressionTrigger;
+use Symfony\Component\Scheduler\Trigger\PeriodicalTrigger;
+use Symfony\Component\Scheduler\Trigger\TriggerInterface;
+
+/**
+ * @experimental
+ */
+class RecurringMessage
+{
+    private function __construct(
+        private readonly TriggerInterface $trigger,
+        private readonly object $message,
+    ) {
+    }
+
+    /**
+     * Uses a relative date format to define the frequency.
+     *
+     * @see https://php.net/datetime.formats.relative
+     */
+    public static function every(string $frequency, object $message, \DateTimeImmutable $from = new \DateTimeImmutable(), ?\DateTimeImmutable $until = new \DateTimeImmutable('3000-01-01')): static
```

Comment:
this is broken if calling `ChildRecurringMessage::every()` as it will still return an instance of RecurringMessage, not of ChildRecurringMessage (due to `new self`).

The private constructor does not really allow to instantiate child classes though (and using `new static()` would require making the constructor final (or making the constructor signature part of an implemented interface) to forbid it to change signature.

I suggest keeping `: self` on those 2 methods, and replacing `::trigger` by a public constructor instead (child classes could then extend the constructor safely).

Alternatively, as this object is just a VO combining the TriggerInterface and the message object, we would keep it final (tests could create an actual instance of that VO)